lib/pack_rb/sub_commands.rb

```diff
@@ -10,16 +10,63 @@ class SubCommands
     include PackRb::SubCommands::Inspect
     include PackRb::SubCommands::Validate
 
+    # Execute a Packer command via {#run_cmd_stream_output}
+    #
+    # @param opts [Hash] options passed to the Packer class
+    #
+    # @return [Array] - stdout [String], stderr [String], exit code [Fixnum]
     def execute(opts)
       cmd = opts[:cmd]
       tpl = opts[:tpl]
 
-      out, err, status = Open3.capture3("#{cmd} -", stdin_data: tpl)
+      out, err, status = run_cmd_stream_output("#{cmd} -", tpl)
+      # rubocop:disable Style/RedundantReturn
+      return [out, err, status]
+      # rubocop:enable Style/RedundantReturn
+    end
 
-      if status.exitstatus != 0
-        puts out if out
-        puts err if err
+    # popen3 wrapper to simultaneously stream command output to the
+    # appropriate file descriptor, and capture it.
+    #
+    # @param cmd [String] command to run
+    # @param tpl [String] the template content
+    # @return [Array] - stdout [String], stderr [String], exit code [Fixnum]
+    def run_cmd_stream_output(cmd, tpl)
```

Comment:
This method should be private. Or pulled into a separate entity where it can be tested as part of that entities public API.